### ISSUE TITLE
Documentation System Improvements

### ISSUE THREAD

**Issue by aeslaughter** [idaholab/moose#8329]
- [ ] Make the new site look correct on mobile device
- [ ] Make the search bar actually search (@dschwen I would appreciate help with this, I can't figure it out.)
- [x] Make the copy code button better
- [ ] Create tests for all markdown extensions and latex conversion items
- [ ] Create link checker (this will likely need to be done after a merge, because some links might not exist until after a merge)
- [x] Clean up code in NavigationNode and utilize the same code path for latex/pdf
- [x] Prevent MathJax from working in code areas
- [ ] Add documentation
- [ ] Auto link should look relative to the md file first, then search the database
- [ ] Need to handle pulling in documentation from submodules in an intelligent and automatic manner

**Comment by aeslaughter** [comment]
To add documentation to MOOSE:

1. Build modules-opt
2. Edit `moose/moosedocs.yml` and remove an item from the list of hidden systems.
3. Run the generate command:
    ```
    cd ~/projects/moose/docs
    ./moosedocs.py generate
    ``` 
    You will see a list of errors that you need to correct: missing markdown and class descriptions.
3. Add the class descriptions and re-compile modules-opt
4. Run the serve command and start writing markdown in the newly generated files:
    ```
    cd ~/projects/moose/docs
    ./moosedocs.py serve
    ``` 
    Navigate to 127.0.0.1:8000 on a browser window.
5. When you are done you should be able to run the 'check' command without errors.
    ```
    cd ~/projects/moose/docs
    ./moosedocs.py check
    ```

**Comment by aeslaughter** [comment]
@idaholab/moose-team Reference this issue when adding documentation, see my [comment](https://github.com/idaholab/moose/issues/8329#issuecomment-273871341) above for quick instructions of the steps you should take.

**Comment by friedmud** [comment]
Ok - so my current issue with the documentation has to do with what's currently listed under the `Documentation` tab.

`Systems`, `Interfaces` and `Modules` won't mean anything to most people.

We need something like `Application Use`, `Application Development`, `MOOSE Development`... or maybe just take out the dropdown for Documentation... and just have it be a link that goes to something that looks more like: http://mooseframework.org/wiki/

**Comment by friedmud** [comment]
Also: there's no reason for `Utilities` to have such a prominent spot on the website.  0.0001% of visits to the site will be so that someone can look at the meta documentation or inquire about the memory logger.

**Comment by aeslaughter** [comment]
@friedmud You can change the top navigation to whatever you want by editing this file: https://github.com/idaholab/moose/blob/devel/docs/navigation.yml

**Comment by permcody** [comment]
Fortunately all of the layout and menus are easily changed. Let's put some
thought into what we want to see. I really want a place to put high level
design information too.

**Comment by friedmud** [comment]
On the individual documentation pages there is still a bit too much wasted whitespace... and the fonts are still too large.  You have to scroll way too much.  Try looking at them on a laptop with a fairly low resolution.

Also: the `EDIT MARKDOWN` button needs to be removed from the sidebar and placed as just a link somewhere at the bottom of each page.  And rename to it `Edit This Documentation` or something clearer.

Along with that: can we change the order of the sidebar so that the "section names" come first... and then the buttons for Doxygen, Header and Source.  Maybe change "Source" to "Source (.C)" or something so it's clearer?

**Comment by friedmud** [comment]
@aeslaughter cool - I'm putting this up for discussion though.  I can change it if there's consensus.  I'm just spitballing here.  We need to decide what we want to see.

**Comment by friedmud** [comment]
@aeslaughter : would you rather I go ahead and change stuff and put up a PR to discuss my changes?  I can do that.

**Comment by friedmud** [comment]
BTW: This is all superficial and organization stuff... I really like the way this shaping up!  Just need to hone in on how we want to present the information.

**Comment by aeslaughter** [comment]
If you want to stylize or change menus go crazy, I am over it.

Moving and renaming the Doxygen, etc. buttons should also be trivial, this is just html template stuff. 

Separating the Edit button to the bottom will take a little bit of python, right now those buttons come out as a list that I iterate over in the html template. I would just need to make a different function for the edit button, nothing difficult.

**Comment by friedmud** [comment]
When trying to build the documentation I get:

```
Traceback (most recent call last):
  File "./moosedocs.py", line 17, in <module>
    import MooseDocs
  File "/Users/gastdr/projects/moose/python/MooseDocs/__init__.py", line 8, in <module>
    import extensions
  File "/Users/gastdr/projects/moose/python/MooseDocs/extensions/__init__.py", line 1, in <module>
    from MooseMarkdown import MooseMarkdown
  File "/Users/gastdr/projects/moose/python/MooseDocs/extensions/MooseMarkdown.py", line 22, in <module>
    from MooseBibtex import MooseBibtex
  File "/Users/gastdr/projects/moose/python/MooseDocs/extensions/MooseBibtex.py", line 6, in <module>
    from pybtex.plugin import find_plugin
ImportError: No module named pybtex.plugin
```

What package do I need to install?  I tried `conda install pybtex`... but that didn't work.  Do you have a full list of the packages we need to install?

**Comment by friedmud** [comment]
Here's what I've done so far to get it working:

```bash
sudo pip install pybtex
sudo pip install markdown-include
sudo conda install beautifulsoup4
sudo pip install python-markdown-math
```

Currently stuck on `git lfs`.... apparently my git (2.5.4) doesn't support LFS?

**Comment by friedmud** [comment]
Ok downloaded git LFS from here: and ran the `install.sh` script.

(Just trying to document this FFR)

**Comment by aeslaughter** [comment]
You probably also need jinja2, if you haven't figured that out already. The lastest package has everything you need.

**Comment by aeslaughter** [comment]
@friedmud I will move the markdown link to the bottom, I am going to change a few other things around for @dschwen.

**Comment by friedmud** [comment]
Sounds good

**Comment by aeslaughter** [comment]
When building an application that is pulling in docs from a submodule, the lookup location is not looking in the correct location.
```
A MOOSE generated stub page for /Adaptivity/Markers/ComboMarker exists, but no content was added. Add documentation content to /Users/schwd/Programs/INL/marmot/doc/content/systems/Adaptivity/Markers/framework/ComboMarker.md.
```

**Comment by bwspenc** [comment]
I'm not sure if this is the right place to put this, but I've found the Google-based search on the website to be pretty frustrating and a big step backward from the previous search capability in my opinion.  It used to be that if I put in the name of an object, it took me right to the Markdown-based page that describes it.  Now I just get a bunch of links to Doxygen pages and broken links.

For example, I just searched for 'PenetrationAux' on the mooseframework.org.  None of the hits on the first screen take me to the Markdown based page for that class, and a whole bunch of the hits take me to broken links.
